frontend/static/js/components/profilePicture.js

```diff
@@ -0,0 +1,46 @@
+import ProfilePictureService from '../services/profilePictureService.js';
+
+const getProfilePicture = () => {
+	const userIdStr = localStorage.getItem('user_id');
+	if (!userIdStr) {
+		throw new Error("User ID is not found in local storage");
+	}
+
+	const userId = parseInt(userIdStr, 10);
+	if (isNaN(userId)) {
+		throw new Error("User ID is not a valid number");
+	}
+
+	const img = document.createElement('img');
+	var profilePictureService = new ProfilePictureService();
+
+	profilePictureService.getRequest(userId)
+	.then(response => {
+		if (!response.ok) {
+			if (response.status === 404) {
+				return '';
+			}
+			else {
+				throw new Error(`Error: ${response.status}`);
+			}
+		}
+
+		return response.json();
+	})
+	.then(responseBody => {
```

Comment:
Mmmeh, we are now using a promise here since the async would bleed the try catching into the sidebar and just felt wrong. Hopefully its okay to do the promise here for the sake of I just dont want to go back and fight this :D

Reply:
It's fine for now, I would say. I am not 100% I understand what you mean with bleeding into the side bar, but either way we'll figure it out once the proper loading of the image everywhere is in place to have a unified handling of that

Reply:
I have started working on a unified error handling thing for all kinds of errors coming in (hence also the above comment regarding the error messages in responses) and in an optimal world we should be able to handle this elegantly

Reply:
Yeah, honestly Im kind of embarrased I could not get it to work properly and that is mostly just me not being sure what would work but Im hoping we can go forward with this being its just here in this function and fix it later together or smt - or then just let it be and pass :D I trust you though so if you're cool with this, nice. If not, lets check it out futher. I am making changes in the next PR too so maybe its premature to change at this point.

Reply:
Guuuurl, you should have a look at the fucked up shit I did in my PR just trying to keep the different modals independent from their surroundings and to keep them as modular as I could xD